components/core/src/clp/streaming_archive/writer/Archive.cpp

```diff
@@ -593,16 +589,12 @@ void Archive::close_segment_and_persist_file_metadata(
 
     for (auto file : files) {
         file->mark_as_in_committed_segment();
+        m_files_written.emplace_back(file);
```

Comment:
just wonder, can we use something like m_files_written.insert(files.begin(), files.end())?

Reply:
Can't say for sure which one is better, so I am fine to keep the emplace.
I do notice that emplace construct the object in place and insert copies the object. but in our case, I guess emplace will also somehow have to copy via the copy constructor.
Anyway, I don't think the performance really matter here. 

@LinZhihao-723 do you have any insights?

Reply:
Ture, you are right. my bad